File: tests/fixtures/sequelize-6/models/post.ts
import { Model, DataTypes } from 'sequelize';

class Post extends Model {
  static associate(models: any) {
    Post.belongsTo(models.User, { foreignKey: 'userId' });
  }
}

Post.init({
  title: { type: DataTypes.STRING, allowNull: false },
  body: DataTypes.TEXT,
  userId: { type: DataTypes.INTEGER, references: { model: 'users', key: 'id' } },
  published: { type: DataTypes.BOOLEAN, allowNull: false },
}, {
  sequelize,
  tableName: 'posts',
  timestamps: true,
});

export default Post;

File: tests/fixtures/python-project/myapp/models/post.py
"""Post model."""

from .base import BaseModel
from .user import User


class Post(BaseModel):
    """A blog post."""

    def __init__(self, title: str, author: User):
        self.title = title
        self.author = author

    def save(self):
        pass

    def delete(self):
        pass
